### PR TITLE
[Merged by Bors] - doc: merge mathlib3 and mathlib4 `README`s

### PR DESCRIPTION
**Pull request by j-loreaux**
This tries to merge the `README`s for mathlib3 and mathlib4 in a sensible way. Some irrelevant material about `leanproject` was removed, as well as a few mathlib3-specific items. Some of these should be replaced by mathlib4 counterparts in the future.

A section has been added about transitioning from Lean 3 to Lean 4 linking to the survival guide and running mathport on a project depending on mathlib.

---

some things that were deleted but need replacing in the future:

from the documentation section:

  In addition to the pages generated for each file in the library, the docs also include pages on:
  - [tactics](https://leanprover-community.github.io/mathlib_docs/tactics.html),
  - [commands](https://leanprover-community.github.io/mathlib_docs/commands.html),
  - [hole commands](https://leanprover-community.github.io/mathlib_docs/hole_commands.html), and
  - [attributes](https://leanprover-community.github.io/mathlib_docs/attributes.html).
- A couple of [tutorial Lean files](docs/tutorial/)

from the guidelines section:

 - The [commit naming conventions](https://github.com/leanprover-community/lean/blob/master/doc/commit_convention.md)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,77 @@
...
-* To obtain precompiled `olean` files, run `lake exe cache get`. (Skipping this step means the next step will be very slow.)
-* To build `mathlib4` run `lake build`.
-* To build and run all tests, run `make`.
-* You can use `lake build Mathlib.Import.Path` to build a particular file, e.g. `lake build Mathlib.Algebra.Group.Defs`.
-* If you added a new file, run the following command to update `Mathlib.lean`
+Mathlib has the following guidelines and conventions that must be followed
 
-  ```shell
-  find Mathlib -name "*.lean" | env LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
-  ```
```
Comment (ADedecker) [inline README.md:44]: This command should be added to the contributing guidelines, I'd say we should keep it their (in the "contributing" section) for now.

**Comment by ADedecker** [comment]
Thanks!

bors d+

File: README.md (hunk 1 above)
Comment (j-loreaux) [reply README.md]: good point, I just added that entire bulleted list back under `Contributing`

**Comment by j-loreaux** [comment]
bors merge

**Comment by j-loreaux** [comment]
trying again?

bors merge

**Pull request closed without merging**